JitsiConference.js

```diff
@@ -72,6 +67,61 @@ function JitsiConference(options) {
 }
 
 /**
+ * Initializes the conference object properties
+ * @param options overrides this.options
+ */
+JitsiConference.prototype._init = function (options) {
+    if(!options)
+        options = {};
+    if(!this.options) {
+        this.options = options;
+    } else {
+        // Override config options
+        var config = options.config || {};
+        for(var key in config)
+            this.options.config[key] = config[key] || this.options.config[key];
+    }
+
+    // Override connection and xmpp properties (Usefull if the connection
+    // reloaded)
+    this.connection = options.connection || this.connection;
+    this.xmpp = this.connection.xmpp;
+    this.retries++;
+    this.room = this.xmpp.createRoom(this.options.name, this.options.config,
+        this.settings, (this.retries < 4 ? 3 : null));
+
+    //restore previous presence options
+    if(options.roomState) {
+        this.room.loadState(options.roomState);
+    }
+    this.room.updateDeviceAvailability(RTC.getDeviceAvailability());
+}
+
+/**
+ * Reloads the conference
+ * @param options {object} options to be overriden
+ */
+JitsiConference.prototype._reload = function (options) {
+    var roomState = this.room.exportState();
+    if(!options)
+        options = {};
+    options.roomState = roomState;
+    this.leave(true);
+    this._reinitialize(options);
+}
+
+/**
+ * Reinitializes JitsiConference instance
+ * @param options {object} options to be overriden
+ */
+JitsiConference.prototype._reinitialize = function (options) {
+    this._init(options || {});
+    this.eventManager.setupChatRoomListeners();
+    this.eventManager.setupStatisticsListeners();
+    this.join();
```

Comment:
I find the _reinitialize method hard to understand:

1. I saw that it's calling this.join & I was initially like "How is it sure that it needs to join?". Then I saw that it was called from a method which does leave first. The leave & join could've been kept in 1 place to aid readability:
````javascript
this.leave(true);
_reinitialize(options);
this.join();
````

Or the prerequisite of the method could be described in a comment.

2. Afterwards i.e. after this.join is not here, the source code could be reduced & cleaned up by eliminating duplicate logic between the constructor & _reinitialize:

- Why init(options || {}) in _reinitialize given that the constructor does init(options) only and init checks whether options is falsy?

- One could observe that both this.eventManager.setupChatRoomListeners(); & this.eventManager.setupStatisticsListeners(); need to be executed after this.init(options). In the constructor the same is being done anyway, it's just split differently (because of _setupListeners). Obviously, this.eventManager.setupRTCListeners(); & setupXMPPListeners need only be called when the respectively involved objects are changed (currently only at construction time). I can then relatively easily image a single _init method & no _reinitialize method. Anyway, that's a point about reducing source code in order to facilitate maintenance so I'm not sure whether it agrees with lib-jitsi-meet's policy.